wait-for-http/.gitlab-ci.yml

```
@@ -0,0 +1,85 @@
+.change_rules_wait_for_http:
+  rules:
+    - if: '$CI_COMMIT_BRANCH == "master"'
+    - &changes_wait_for_http
+      changes:
+        - wait-for-http/**/*
+
+Lint wait-for-http:
+  stage: "Quality"
+  extends: .change_rules_wait_for_http
+  image: hadolint/hadolint:v1.18.2-debian
+  script:
+    - hadolint ./wait-for-http/Dockerfile
+
+Build wait-for-http:
+  stage: "Build"
+  extends:
+    - .base_register_to_gitlab_stage
+    - .change_rules_wait_for_http
+  needs:
+    - Lint wait-for-http
+  variables:
+    CONTEXT: wait-for-http
+    IMAGE_NAME: ${CI_REGISTRY_IMAGE}/wait-for-http
+
+Test wait-for-http 1:
+  needs:
+    - Lint wait-for-http
+    - Build wait-for-http
+  stage: "Test"
+  extends: .change_rules_wait_for_http
+  image:
+    name: "${CI_REGISTRY_IMAGE}/wait-for-http:${CI_COMMIT_SHA}"
+    entrypoint: ["/bin/sh"]
+  variables:
+    RETRIES: 5
+  # expect wait to fail and a retry duration >= 25secs (5*5)
+  script: |
+    start=`date +%s`
+    ! /bin/wait-for-http http://invalid
+    end=`date +%s`
+    runtime=$((end-start))
+    [[ $runtime -gt 25 ]]
```

25s is a long time waiting for failure nan ?

```suggestion
  variables:
    RETRIES: 1
  # expect wait to fail and a retry duration >= 5secs
  script: |
    start=`date +%s`
    ! /bin/wait-for-http http://invalid
    end=`date +%s`
    runtime=$((end-start))
    [[ $runtime -gt 5 ]]
```

1s is enough to prove that it works no ?

---

there is 5 retries of 5s each so the test prove it retried long enough ? 2 at least should be enough btw

---

Let's change the interval to make it possible :)

---

its now `WAIT_FOR_INTERVAL`

---

```suggestion
  variables:
    WAIT_FOR_RETRIES: 5
  # expect wait to fail and a retry duration >= 25secs (5*5)
  script: |
    start=`date +%s`
    ! /bin/wait-for-http http://invalid
    end=`date +%s`
    runtime=$((end-start))
    [[ $runtime -gt 5 ]]
```

---

WAIT_FOR_RETRIES * WAIT_FOR_INTERVAL = 25

---

Nieh... 
I mess it up with the gh suggestion let's do it again